From c206b13beba9718c7a3a71c8fe4ba63c2e79b632 Mon Sep 17 00:00:00 2001
From: qianmenx <qianx.meng@intel.com>
Date: Thu, 25 Jul 2019 10:40:28 +0800
Subject: [PATCH 2350/2367] media: intel-ipu4: resolve compile error with
 allyesconfig

Change-Id: If4c714c4adb771452417e943de1410875cc06e25
Tracked-On: #H1507330328
Tracked-On: #PKT-2935
Signed-off-by: qianmenx <qianx.meng@intel.com>
(cherry picked from commit efc0a885761ccf7c3ea897914ecbe772c4cd9772)
Signed-off-by: Meng Wei <wei.meng@intel.com>
Signed-off-by: Ong Yang Yang <yang.yangx.ong@intel.com>
---
 drivers/media/i2c/dw9714.c               | 1 +
 drivers/media/pci/intel/ipu-isys-video.c | 4 ++++
 include/media/as3638.h                   | 6 ++++++
 include/uapi/linux/ipu-isys-isa-fw.h     | 6 ++++++
 include/uapi/linux/ipu-psys.h            | 4 ++++
 5 files changed, 21 insertions(+)

diff --git a/drivers/media/i2c/dw9714.c b/drivers/media/i2c/dw9714.c
index 1d244d1..384b48d 100644
--- a/drivers/media/i2c/dw9714.c
+++ b/drivers/media/i2c/dw9714.c
@@ -6,6 +6,7 @@
  *
  */
 
+#include <linux/version.h>
 #include <linux/delay.h>
 #include <linux/gpio.h>
 #include <linux/i2c.h>
diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 776ed0d..a1ffa29 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -1749,7 +1749,11 @@ static const struct v4l2_ioctl_ops ioctl_ops_splane = {
 
 static const struct v4l2_ioctl_ops ioctl_ops_mplane = {
 	.vidioc_querycap = ipu_isys_vidioc_querycap,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+	.vidioc_enum_fmt_vid_cap = ipu_isys_vidioc_enum_fmt,
+#else
 	.vidioc_enum_fmt_vid_cap_mplane = ipu_isys_vidioc_enum_fmt,
+#endif
 	.vidioc_g_fmt_vid_cap_mplane = vidioc_g_fmt_vid_cap_mplane,
 	.vidioc_s_fmt_vid_cap_mplane = vidioc_s_fmt_vid_cap_mplane,
 	.vidioc_try_fmt_vid_cap_mplane = vidioc_try_fmt_vid_cap_mplane,
diff --git a/include/media/as3638.h b/include/media/as3638.h
index 8e847ef..6fd6b69 100644
--- a/include/media/as3638.h
+++ b/include/media/as3638.h
@@ -4,6 +4,12 @@
 #ifndef __AS3638_H__
 #define __AS3638_H__
 
+#ifdef __KERNEL__
+#include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
+
 #define AS3638_NAME			"as3638"
 #define AS3638_I2C_ADDR			0x30
 
diff --git a/include/uapi/linux/ipu-isys-isa-fw.h b/include/uapi/linux/ipu-isys-isa-fw.h
index f0914bc..3a4ee6d 100644
--- a/include/uapi/linux/ipu-isys-isa-fw.h
+++ b/include/uapi/linux/ipu-isys-isa-fw.h
@@ -4,6 +4,12 @@
 #ifndef IPU_ISYS_ISA_FW_H
 #define IPU_ISYS_ISA_FW_H
 
+#ifdef __KERNEL__
+#include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
+
 #define ia_css_terminal_offsets(pg)			\
 	((uint16_t *)((void *)(pg) +			\
 		      (pg)->terminals_offset_offset))
diff --git a/include/uapi/linux/ipu-psys.h b/include/uapi/linux/ipu-psys.h
index 0f26dbf..c20da18 100644
--- a/include/uapi/linux/ipu-psys.h
+++ b/include/uapi/linux/ipu-psys.h
@@ -4,7 +4,11 @@
 #ifndef _UAPI_IPU_PSYS_H
 #define _UAPI_IPU_PSYS_H
 
+#ifdef __KERNEL__
 #include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
 
 struct ipu_psys_capability {
 	uint32_t version;
-- 
2.7.4

